config/src/test/java/beanconfig/ObjectsConfig.java

```diff
@@ -24,6 +26,14 @@ public String getOptionalValue() {
         public void setOptionalValue(String optionalValue) {
           this.optionalValue = optionalValue;
         }
+
+        public String getNullableValue() {
+            return nullableValue;
+        }
+
+        public void setNullableValue(String nullableValue) {
+            this.nullableValue = nullableValue;
```

Comment:
I think (could be wrong) that we aren't testing the behavior where the bean constructor sets a non-null value on the optional field, and then either we do not have it in the config, or it's set to null in the config. Intuitively I would think it should be overwritten with null only if explicitly set in the config?